pkg/apis/core/types.go

```diff
@@ -475,6 +475,25 @@ type PersistentVolumeClaimSpec struct {
 	// (Beta) Using this field requires the AnyVolumeDataSource feature gate to be enabled.
 	// +optional
 	DataSourceRef *TypedLocalObjectReference
+	// (Alpha) Using this field requires the CrossNamespaceSnapshots feature gate to be enabled.
+	// +optional
+	DataSourceRef2 *TypedCrossNamespaceObjectReference
```

Comment:
I was just discussing with some other api reviewers on the new field. One thing I learned is that the go type can change as long as the original fields serialize the same. So that means we could potentially add the namespace field to the existing `dataSourceRef` instead of creating a new field. The main concern I have about this is data safety because drivers using older sidecars will ignore the namespace field, and could end up using the local reference with the same name.

However considering that this field is still beta, we have some leeway here to deprecate those sidecars, put a max k8s version on them, and have folks upgrade to the newer sidecars. I think from an API perspective that would be cleaner than having a new field. WDYT?

Reply:
I think Good Idea. I also think it would be better from a UX point of view to be able to use existing fields.
But, I am concerned about the following two points

1. Since the data is copied between DataSourceRef and DataSource, I think DataSource should also be changed to the same type. Is this also acceptable?

2. When a request with namespace set to DataSourceRef comes, is it okay to simply copy to DataSource?
For example, if the feature gate is disabled after being copied with the feature gate enabled, won't the snapshot with the same name in the namespace of the PVC be specified by mistake instead of the specified snapshot?

Reply:
We should be really careful about making any changes to the DataSource field which is already GA. Many products are already depending on this field. We should not require them to make any changes.

For the DataSourceRef field which is Beta, I'm fine changing it as long as it does not break DataSource.

Reply:
Conclusion was that we only add the Namespace field to DataSourceRef which is beta, and don't modify DataSource which is GA.